x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/indexlifecycle/LifecycleAction.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.protocol.xpack.indexlifecycle;
+
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+
+/**
+ * Parent class for the client-side ilm actions for the {@link NamedXContentRegistry}
```

Comment:
nit: I think this should be worded differently given its public:
```
Marker interface for index lifecycle management actions
```
Reasons:
* This is a marker interface not a parent class
* The fact that its a client side copy is an implementation detail that client users should not need to know about
* The fact that we use it to have a common category class for the NamedXContentRegistry is an implementation detail I don't think clients should need to consider